utils/utils.c

```diff
@@ -392,3 +392,95 @@ char * strjoin(char *left, char *right, char *delim)
 	strcpy(new + len - rlen - 1, right);
 	return new;
 }
+
+#define QUOTE '\''
+#define DQUOTE '"'
+
+static int setargs(char *args, char **argv)
+{
+	int count = 0;
+
+	/* ignore start spaces */
+	while (isspace(*args))
+		++args;
+
+	while (*args) {
+		/* consider quotes and update argv */
+		if (*args == QUOTE) {
+			if (argv)
+				argv[count] = ++args;
+			while (*args != QUOTE)
+				++args;
+			if (argv)
+				*args = ' ';
+		}
+		else if (*args == DQUOTE) {
+			if (argv)
+				argv[count] = ++args;
+			while (*args != DQUOTE)
+				++args;
+			if (argv)
+				*args = ' ';
+		}
+		else if (argv) {
+			argv[count] = args;
+		}
+		/* read characters until '\0' or space */
+		while (*args && !isspace(*args))
+			++args;
+		/* set '\0' rather than space */
+		if (argv && *args)
+			*args++ = '\0';
+		/* ignore margin spaces */
+		while (isspace(*args))
+			++args;
+		/* count up argument */
+		count++;
+	}
+
+	return count;
+}
+
+#undef QUOTE
+#undef DQUOTE
+
+char **parse_cmdline(char *cmd, int *argc)
```

Comment:
Sorry for bothering you.. but I think it'd be better placing it in the .c file.  Also please add a similar comment for free_parsed_cmdline() too.